src/sentry/utils/http.py

```diff
@@ -209,31 +212,38 @@ def is_valid_origin(origin, project=None, allowed=None):
     return False
 
 
-def is_valid_ip(ip_address, project):
+def is_valid_for_processing(value, filter_type, project):
```

Comment:
i think it'd be better for testing/readability to just leave this function as `is_valid_ip` and instead add additional functions to check error messages and releases. then you can just call each of them from `should_filter`.